### PR TITLE
Add support for markers and completion items

### PR DESCRIPTION
**Pull request by LePtitDev**
* Add markers and quick fixes  
  image
* Add completion item  
  image

### CODE REVIEW

File: BlazorMonaco/Bridge/Data.cs [hunk 1]
```diff
@@ -103,4 +103,81 @@ public class OverviewRulerPosition
...
+        public string Text { get; set; }
+
+        public Range Range { get; set; }
+
+        public EndOfLineSequence? Eol { get; set; }
+    }
+
+    public class CompletionItem
+    {
+        public string Label { get; set; }
```
Comment (KTAlcoda) [inline BlazorMonaco/Bridge/Data.cs:171]: The Label can also be an object implementing the interface CompletionItemLabel. (See https://github.com/microsoft/monaco-editor/blob/22698f2/website/typedoc/monaco.d.ts#L6175)

**Review by KTAlcoda — COMMENTED** [review]
Looks good overall. Did not test these features extensively. Would like to see them merged. What is the status?

File: BlazorMonaco/wwwroot/jsInterop.js [hunk 2]
```diff
@@ -843,3 +853,56 @@ window.blazorMonaco.editor = {
...
+                    });
+                });
+            }
+        });
+    },
+
+    registerCompletionItemProvider: function (language, provider) {
+        monaco.languages.registerCompletionItemProvider(language, {
+            provideCompletionItems: (model, position, context, token) => {
+                let task = provider.invokeMethodAsync("Invoke", decodeURI(model.uri.toString()), position);
```
Comment (KTAlcoda) [inline BlazorMonaco/wwwroot/jsInterop.js:895]: Could we also pass the entire model object?

**Comment by serdarciplak** [comment]
I’m currently working on aligning BlazorMonaco’s structure and class names better with MonacoEditor. I’ll add this feature after that is done.

**Comment by KTAlcoda** [comment]
An issue I encountered when playing with the code is the improper disposing of the completionItemProvider. When creating multiple instances accross multiple razor pages of the blazor monaco editor, code completion items get reregistered on every page switch.

There are multiple ways of dealing with this, see for instance https://github.com/microsoft/monaco-editor/issues/2217 or https://github.com/microsoft/monaco-editor/issues/2084.
One option could be to save a reference of the completion item provider and dispose that reference when disposing monaco editor in jsInterop. That is upon call of 'create' or 'dispose'.

**Comment by owaits** [comment]
This PR seems to have been up for a while now, is there any possibility of this feature being merged a published in a nuget update?

**Comment by adam-dot-cohen** [comment]
Any updates on this?

**Comment by serdarciplak** [comment]
This PR is not up to date with v3.0.0 and cannot be merged. I'm making the same changes in the latest version. I'll let you know and close this PR when I'm done.

**Comment by serdarciplak** [comment]
Closing this PR as #99 made the same changes on the up-to-date version and aligned the interface of these methods with the original Monaco Editor.

The changes are released in v3.1.0

**Pull request closed without merging**